### PR TITLE
issue #2859570 Create a generic entity bundle delete form

### PR DESCRIPTION
**Pull request by finnef**

### CODE REVIEW

**Comment by finnef** [comment]
DO: https://www.drupal.org/node/2859570

File: src/Form/CommerceBundleEntityDeleteFormBase.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    /** @var \Drupal\Core\Config\Entity\ConfigEntityType $bundle_entity_type */
+    $bundle_entity_type = $this->entityTypeManager->getDefinition($this->entity->getEntityTypeId());
+    /** @var \Drupal\Core\Entity\ContentEntityType $content_entity_type */
+    $content_entity_type = $this->entityTypeManager->getDefinition($bundle_entity_type->getBundleOf());
+    $usage_count = $this->entityTypeManager->getStorage($content_entity_type->id())
+      ->getQuery()
+      ->condition('type', $this->entity->id())
```
Comment (mglaman) [inline src/Form/CommerceBundleEntityDeleteFormBase.php]: You need to check the content entity type's definition to get the proper bundle key.

**Review by mglaman — CHANGES_REQUESTED** [review]

File: src/Form/CommerceBundleEntityDeleteFormBase.php (hunk 1 above)
Comment (mglaman) [reply src/Form/CommerceBundleEntityDeleteFormBase.php]: Forgot to add: not sre if that's why the functional tests are breaking or not.

**Comment by mglaman** [comment]
Shoot, I thought last change would fix tests. Not sure why, it might just be tweaks to expected error message.

**Comment by finnef** [comment]
I tried this test locally and it succeeds. The Travis CI fails on 
Drupal\Tests\commerce_order\Functional\OrderTypeTest::testDeleteOrderType (line 92)
Why I don't know. Hard to get screenshots of the CI tester I would guess?

**Comment by finnef** [comment]
Found the failure. Changed some test strings to match the form.

**Comment by finnef** [comment]
On my latest patch all tests have passed. I don't know why travis says the checks failed...?

**Comment by mglaman** [comment]
@finnef are you running phpunit locally or run-tests? I've had issues where local with phpunit is fine, but concurrent in run-tests goes nuts.

**Comment by finnef** [comment]
I run phpunit... but my previous comment was about travis: if you check the current 'failed' details, all tests have passed. Strange.

**Comment by mglaman** [comment]
oh, I didn't look. It's coding standards error. I'll just edit in UI for you

```
FILE: ...ommerce/commerce/src/Form/CommerceBundleEntityDeleteFormBase.php
----------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------
 34 | WARNING | [x] A comma should follow the last multiline array item.
    |         |     Found: ) (Drupal.Array.Array.CommaLastItem)
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
Time: 670ms; Memory: 10Mb
```

**Review by mglaman — APPROVED** [review]

**Comment by mglaman** [comment]
\o/ yay

**Pull request merged**